### PR TITLE
fix(ko): Display correct tag for sideloaded images

### PR DESCRIPTION
**Pull request by halvards**
Display the expected tag after successfully building a ko image, for both sideloaded and published images.

Also handles the `tagPolicy.sha256` scenarios.

Fixes: #6835
Tracking: #6041

### CODE REVIEW

File: pkg/skaffold/build/ko/build.go [hunk 1]
```diff
@@ -52,7 +52,13 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, a *latestV1.Artifact
...
 		return "", fmt.Errorf("could not build and publish ko image %q: %w", a.ImageName, err)
 	}
-	fmt.Fprintln(out, imageRef.Name())
+
+	outputRef := ref
+	if strings.HasSuffix(ref, ":latest") {
+		// handle tagPolicy sha256
+		outputRef = imageRef.Name()
+	}
+	fmt.Fprintln(out, outputRef)
```
Comment (briandealwis) [inline pkg/skaffold/build/ko/build.go:61]: We're not outputting the image ref for the other builders.  Why is this necessary?

Comment (halvards) [reply pkg/skaffold/build/ko/build.go:61]: Good question. `ko` itself prints it out at the end, and I've found that helpful, so this was just mimicking that behavior. And when using the Docker builder, the image name is printed at the end, even though Skaffold doesn't do it.

Comment (briandealwis) [reply pkg/skaffold/build/ko/build.go:61]: ok makes sense

File: pkg/skaffold/build/ko/build.go [hunk 2]
```diff
@@ -52,7 +52,13 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, a *latestV1.Artifact
 	if err != nil {
 		return "", fmt.Errorf("could not build and publish ko image %q: %w", a.ImageName, err)
 	}
-	fmt.Fprintln(out, imageRef.Name())
+
+	outputRef := ref
+	if strings.HasSuffix(ref, ":latest") {
```
Comment (briandealwis) [inline pkg/skaffold/build/ko/build.go:57]: So does Ko only do use the digest for :latest?

**Comment by halvards** [comment]
Marking as draft until I have time to get back to this one.

**Comment by halvards** [comment]
Closing in favor of #6928
